sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -192,4 +208,44 @@ public class ChatClient {
             }
         }
     }
+
+    /// Start receiving realtime notifications.
+    /// Call this function before subscribing to any event.
+    public func startRealTimeNotifications() throws {
+        if isRealtimeNotificationsStarted {
+            return
+        }
+
+        isRealtimeNotificationsStarted = true
+        signalingClient.start()
+    }
+
+    /// Stop receiving realtime notifications.
+    /// This function would unsubscribe to all events.
+    public func stopRealTimeNotifications() throws {
+        isRealtimeNotificationsStarted = false
+        signalingClient.stop()
+    }
+
+    /// Subscribe to chat events.
+    /// - Parameters:
+    ///   - event: The chat event to subsribe to.
+    ///   - listener: The listener for the chat event.
+    public func on(
+        event: ChatEventId,
+        listener: @escaping EventListener
+    ) {
+        signalingClient.on(event: event.rawValue, listener: listener)
+    }
+
+    /// Unsubscribe to chat events.
+    /// - Parameters:
+    ///   - event: The chat event to subsribe to.
+    ///   - listener: The listener for the chat event.
+    public func off(
+        event: ChatEventId,
+        listener: @escaping EventListener
+    ) {
+        signalingClient.off(event: event.rawValue, listener: listener)
+    }
```

Comment:
It looks like the underlying client from trouter expects an actual handler for its unregister method so CommunicationSignalingClient is just mirroring that.

@glorialimicrosoft does trouter support subscribing different listeners for the same event?

Reply:
@knvsl there can only be one trouter listener for each path (event).
Currently, we have a communicationListeners set to maintain all the listeners the user has subscribed to.
I suppose we don't need to pass in the listener if we keep a dictionary where the key is the event type and value is the communicationListener, and when user sends a off request for some event, we can locate that listener with the event type, then unregister that listener.

Reply:
I can see that some of this is needed to conform to the underlying trouter implementation, but it seems this facade can be improved for usability.

Reply:
Thanks @glorialimicrosoft, since there can be only one listener then yes I think for ease of use unregistering shouldn't require you to provide the listener again.